pandas/core/groupby/groupby.py

```diff
@@ -2173,6 +2174,9 @@ def pre_processor(vals: np.ndarray) -> Tuple[np.ndarray, Optional[Type]]:
             elif is_datetime64_dtype(vals.dtype):
                 inference = "datetime64[ns]"
                 vals = np.asarray(vals).astype(float)
+            elif is_timedelta64_dtype(vals.dtype):
```

Comment:
this pre-processing seems *very* generic. can we move this to a function (e.g. this is basically cast the results so that I can use them in groupby). Please make an issue / PR for this as a followon.

Reply:
Done, #37505